### PR TITLE
Stop recommending `-jnlpUrl`

### PR DESCRIPTION
**Pull request by basil**
See https://github.com/jenkinsci/remoting/pull/677#discussion_r1364735464; the `-jnlpUrl` form essentially means you don't have to specify `-name` and things like `-webSocket`, `-workDir`, etc but then requires Remoting to query the server for these values. Simpler and less error-prone to just pass in the desired arguments directly without the extra layer of indirection.

Unfortunately, passing in `-name` via a shell can be tricky if the name contains special characters. We tested with a name of ``"foo's `bar"`` (yes, that includes both single and double quotes, backticks, and spaces) on both Unix and Windows to come up with the escaping algorithm. Fortunately we don't need to do deal with most other characters that need escaping in shells due to the restrictions imposed by `Jenkins#checkGoodName`.

The end result is a simpler connection algorithm on the Remoting side that skips an additional request to the `.jnlp` file on the server and instead begins execution directly with the desired set of arguments. This removal of a layer of indirection makes things easier to understand and debug.

### Testing done

Tested the instructions on both Unix and Windows with an agent named ``"foo's `bar"``.

### Proposed changelog entries

- Stop recommending JNLP URL in agent launch instructions.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Comment by basil** [comment]
I don't think the escaping is all that interesting since this is a command the user is expected to copy and paste rather than something automatically executed, but CC'ing the security team in case they are interested.

File: core/src/main/java/hudson/slaves/JNLPLauncher.java
Comment (jglick) [inline core/src/main/java/hudson/slaves/JNLPLauncher.java:218]: We could simply deprecate this option, since it will now have no effect other than changing the recommendation for the command. Compare discussion of `vmargs` in #6543.

File: core/src/main/java/hudson/slaves/JNLPLauncher.java
Comment (jglick) [inline core/src/main/java/hudson/slaves/JNLPLauncher.java:221]: Same here. Also `workDirSettings`. IOW there is no longer a reason to have _any_ configurable parameters on `JNLPLauncher`.

**Review by jglick — APPROVED** [review]
A good start.

**Comment by basil** [comment]
> A good start.

If you think more things can be cleaned up, then by all means, propose some pull requests to clean them up. I will not be taking cleanup suggestions.

File: core/src/main/java/hudson/slaves/JNLPLauncher.java
Comment (daniel-beck) [reply core/src/main/java/hudson/slaves/JNLPLauncher.java:218]: So without this change (including upstream), agents automatically apply configuration changes in Jenkins on reconnection (as the JNLP file has these), while now the agent command line needs to changed? That seems confusing/unexpected as an admin and worth documenting as least IMO (and arguably that was a pretty nice feature to have).

**Comment by daniel-beck** [comment]
Seems fine from security POV, thanks for considering the escaping here.

**Comment by MarkEWaite** [comment]
This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

**Pull request merged**
